### PR TITLE
Add NFT ChangeOwner action tests

### PR DESCRIPTION
**Pull request by gege251**
- Updating plutus-extra to enable payToPKH with datums
- adding tests for ChangeOwner

### CODE REVIEW

File: mlabs/src/Mlabs/EfficientNFT/Token.hs [hunk 1]
```diff
@@ -90,13 +91,21 @@ mkPolicy burnHash previousNft collectionNftP mintAct ctx =
 
     -- Check if the old token is burnt and new is minted with correct name
     checkMintAndBurn nft newPrice newOwner =
-      let oldName = mkTokenName nft
+      let burntMinted =
+            sortOn (\(_, _, amt) -> amt)
```
Comment (zygomeb) [inline mlabs/src/Mlabs/EfficientNFT/Token.hs:95]: as in the other PR, not sure if that's the most efficient way of doing this

File: mlabs/src/Mlabs/EfficientNFT/Token.hs [hunk 2]
```diff
@@ -90,13 +91,21 @@ mkPolicy burnHash previousNft collectionNftP mintAct ctx =
...
+              . filter (\(cs, _, _) -> cs == ownCs)
+              $ Value.flattenValue (txInfoMint info)
+          oldName = mkTokenName nft
           newName = mkTokenName nft {nftId'price = newPrice, nftId'owner = newOwner}
-          validBurn = Value.valueOf (txInfoMint info) ownCs oldName == -1
-          validMint = case filter (\(cs, _, _) -> cs == ownCs) $ Value.flattenValue (txInfoMint info) of
-            [(_, tn, amt)] -> tn == newName && amt == 1
+       in case burntMinted of
+            [(_, oldTokenName, oldTnAmt), (_, newTokenName, newTnAmt)] ->
+              and
```
Comment (zygomeb) [inline mlabs/src/Mlabs/EfficientNFT/Token.hs:102]: should hardcode the conjunction, using 'and' here adds to exunits unfortunately

File: mlabs/src/Mlabs/EfficientNFT/Token.hs (hunk 1 above)
Comment (gege251) [reply mlabs/src/Mlabs/EfficientNFT/Token.hs]: Simplified this

**Comment by gege251** [comment]
> Merging when conflicts get resolved. Not sure why you guys keep updating the old NFT versions :D

Unfortunately updating a dependency breaks the old code, so we have to fix that as well

**Pull request merged**